### PR TITLE
#14346 [DistSQL] Optimize query results of ```SHOW AUTHORITY RULE```

### LINKED ISSUE #14346: [DistSQL] Optimize query results of `SHOW AUTHORITY RULE`
After PR #14260, we can query authority rule through DistSQL:

image

But now anyone can see the passwords of other users, which needs to be improved.

I suggest that when users execute `SHOW AUTHORITY RULE`, only query the user name, not the secret.

### PR DESCRIPTION
**Pull request by CrackerSuperior**
Fixes #14346.

Changes proposed in this pull request:
- Optimize query results of `SHOW AUTHORITY RULE`
- Remove user password from query results
- 

-

### CODE REVIEW

**Comment by CrackerSuperior** [comment]
image
What's this? : ( 
@RaigorJiang

**Comment by RaigorJiang** [comment]
> What's this? : ( 

This is a problem caused by other modifications, we can ignore it for now

**Comment by CrackerSuperior** [comment]
> > What's this? : (
> 
> This is a problem caused by other modifications, we can ignore it for now

ok，:)

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/executor/ShowAuthorityRuleExecutor.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ protected MergedResult createMergedResult() {
         }
         AuthorityRuleConfiguration authorityRuleConfiguration = authorityRuleConfigurationOptional.get();
         List<Object> row = new LinkedList<>();
-        row.add(authorityRuleConfiguration.getUsers().stream().map(each -> each.getGrantee() + ":" + each.getPassword()).collect(Collectors.joining("; ")));
+        row.add(authorityRuleConfiguration.getUsers().stream().map(each -> each.getGrantee() + "").collect(Collectors.joining("; ")));
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/executor/ShowAuthorityRuleExecutor.java:58]: Why is `+ ""` needed here, can it be removed or replaced by toString()?

Comment (CrackerSuperior) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/executor/ShowAuthorityRuleExecutor.java:58]: Of course you can, immediately change to toString()

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**